kobo/apps/trackers/utils.py

```diff
@@ -43,17 +46,36 @@ def update_nlp_counter(
 
     # Update the total counters by the usage amount to keep them current
     kwargs = {}
+    addon_amount_used = 0
     if service.endswith('asr_seconds'):
         kwargs['total_asr_seconds'] = F('total_asr_seconds') + amount
         if asset_id is not None:
             # If we're not updating the catch-all counter, increment any NLP add-ons the user may have
-            PlanAddOn.increment_add_ons_for_user(user_id, 'seconds', amount)
+            remaining = PlanAddOn.increment_add_ons_for_user(
```

Comment:
`increment_add_ons_for_user()` gets a list of relevant addons and, for each addon, checks [is_organization_over_plan_limit()](https://github.com/kobotoolbox/kpi/blob/06b50e2712272d53f92cddfa9bcf4cb43b533613/kobo/apps/organizations/models.py#L77) for that organization. Maybe I'm missing a step somewhere, but it seems to me like this only applies to cases where a user has reached their plan limit at some point in the past. But consider a case where I have a plan limit of 1000 asr minutes and an addon for 2000 additional asr minutes, then I use 2000 asr minutes.  I would expect my addon minutes limits_remaining to go down to 1000. Currently it doesn't change at all. 

If you are curious about this, the [addon tracking tests](kobo/apps/stripe/tests/test_one_time_addons_tracking.py) above can be changed slightly to test it out.

Reply:
Changed `is_organization_over_plan_limit` to `check_usage_exceeds_plan_limit`, which can check against incoming usage.